Extend/Model/PaymentGateway.php

```diff
@@ -119,63 +109,114 @@ public function getRedirectUrls(Session $oSession, string $sShopUrl): Redirect
     /**
      * Executes the transaction through EE
      *
-     * @param float $fAmount
-     * @param Order $oOrder
+     * @param Basket $oBasket
+     * @param Order  $oOrder
      *
-     * @return FailureResponse|InteractionResponse|Response|SuccessResponse
+     * @return Transaction
      *
      * @throws Exception
      *
-     * @SuppressWarnings(PHPMD.Coverage)
      * @SuppressWarnings(PHPMD.CyclomaticComplexity)
      * @SuppressWarnings(PHPMD.ElseExpression)
      * @SuppressWarnings(PHPMD.ExcessiveMethodLength)
      */
-    public function makeTransaction(float $fAmount, Order $oOrder): Response
+    public function createTransaction($oBasket, $oOrder)
     {
         $sShopUrl = $this->getConfig()->getCurrentShopUrl();
+
         $oSession = $this->getSession();
-        $oBasket = $oSession->getBasket();
-        $sPaymentMethod = $oBasket->getPaymentId();
 
-        $oRedirect = self::getRedirectUrls($oSession, $sShopUrl);
-        $oPaymentMethod = Payment_Method_Factory::create($sPaymentMethod);
-        $oTransactionService = new TransactionService($oPaymentMethod->getConfig(), $this->_oLogger);
+        /**
+         * @var $oPayment Payment
+         */
+        $oPayment = oxNew(Payment::class);
+        $oPayment->load($oBasket->getPaymentId());
+
+        $sSid = $oSession->sid(true);
```

Comment:
if you have a helper you can use the same method as above in another class for this

Reply:
👍